llap-server/src/main/resources/llap-daemon-log4j2.properties

```diff
@@ -81,23 +81,21 @@ appender.query-routing.type = Routing
 appender.query-routing.name = query-routing
 appender.query-routing.routes.type = Routes
 appender.query-routing.routes.pattern = $${ctx:queryId}
-#Purge polciy for query-based Routing Appender
-appender.query-routing.purgePolicy.type = LlapRoutingAppenderPurgePolicy
-# Note: Do not change this name without changing the corresponding entry in LlapConstants
-appender.query-routing.purgePolicy.name = llapLogPurgerQueryRouting
+#Purge policy for query-based Routing Appender
+appender.query-routing.purgePolicy.type = IdlePurgePolicy
+appender.query-routing.purgePolicy.timeToLive = 5
```

Comment:
If 5 sec pass without a write then the appender is deleted and the file is closed. The appender/file will reopen again when logging activity comes back. Do you think this might be a problem?

Reply:
Yes. 5 sec is too low. I would keep give it atleast 30 or 60s. Also I remember the log router will rename the files with .done suffix to mark the completion of query and hence the log file. Does it still happen? If we no longer follow the filename convention it will be worth checking if other consumers are expecting files with .done suffix (yarn log aggregation, twz ui)

Reply:
@prasanthj I increased the `timeToLive` for production to 60s and made appropriate changes to keep the `.done` when the query completes. I tested the changes on the cluster and things seem to work properly (no descriptor leak, yarn log aggregation working normally). Can you have another look when you find some time?